packages/flutter/lib/src/material/text_field.dart

```diff
@@ -244,6 +324,9 @@ class _TextFieldState extends State<TextField> {
     final TextStyle style = widget.style ?? themeData.textTheme.subhead;
     final TextEditingController controller = _effectiveController;
     final FocusNode focusNode = _effectiveFocusNode;
+    final List<TextInputFormatter> formatters = widget.inputFormatters ?? <TextInputFormatter>[];
+    if (widget.maxLength != null && widget.maxLengthEnforced)
```

Comment:
if you assert, in the constructor, that `!(maxLength == null && maxLengthEnforced)`, then you can just check maxLengthEnforced here.

Reply:
You could have the constructor default the maxLengthEnforced argument to null, then set the property to true if maxLength is non-null and maxLengthEnforced is null, and false otherwise.